Adopting  Security Groups API V3

radoslav-tomov opened the pull request:
Adopting Security Groups API V3 API. Change includes  implementation, unit and integration tests for the following operations:

- [ Create Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#create-a-security-group)
- [ Get Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#get-a-security-group)
- [ List Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#list-security-groups)
- [ Update Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#update-a-security-group)
- [Delete Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#delete-a-security-group)
- [Bind Running Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#bind-a-running-security-group-to-spaces)
- [Bind Staging Security group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#bind-a-staging-security-group-to-spaces)
-  [Unbind Running Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#unbind-a-running-security-group-to-spaces)
- [Unind Staging Security group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#unbind-a-staging-security-group-to-spaces)
-  [List Running Security Group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#list-a-running-security-group-to-spaces)
- [List Staging Security group](https://v3-apidocs.cloudfoundry.org/version/3.140.0/index.html#list-a-staging-security-group-to-spaces)

PR ended up a bit on the large side, but it is mostly boilerplate, so I decided not to try to brake it further apart.

anthonydahanne commented on integration-test/src/test/java/org/cloudfoundry/client/v3/SecurityGroupsTest.java:
```diff
@@ -0,0 +1,253 @@
+        private String securityGroupName;
+
+        @Before
+        public void settup() {
```
setup

anthonydahanne commented on cloudfoundry-client-reactor/src/main/java/org/cloudfoundry/reactor/client/v3/securitygroups/ReactorSecurityGroupsV3.java:
```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
```
2013-2023?

radoslav-tomov replied:
@anthonydahanne  Thanks for finding time to check it out. Will adopt the feedback.

anthonydahanne reviewed (changes requested):
all tests pass; I've had a quick look, please fix obvious issues (cf screenshots)

thank you
<img width="1206" alt="Screenshot 2023-10-04 at 17 34 26" src="https://github.com/cloudfoundry/cf-java-client/assets/60938/56cbad31-646c-4d84-bfac-997a29e96cdb">
<img width="1164" alt="Screenshot 2023-10-04 at 17 34 13" src="https://github.com/cloudfoundry/cf-java-client/assets/60938/8e04a4c0-8475-4803-a802-2f987229124d">

radoslav-tomov commented:
@anthonydahanne  I've removed the unused imports and reformatted via the standard google eclipse style formatter. Fixed the year in the license comment and added it where it was missing. Unfortunately I don't have InteliJ license, so I might have missed some warnings. Let me know if this is the case.

anthonydahanne commented:
Hello @radoslav-tomov !
It's better, thank you, but still a lot of  useless reformating.

But I don't blame you at all; the project should have a linter in place; working on that!

anthonydahanne commented:
was included and merged with this PR https://github.com/cloudfoundry/cf-java-client/pull/1215

pivotal-david-osullivan commented:
Closing as this work was included in #1215
